executor/runtime/docker/docker.go

```diff
@@ -2450,40 +2452,53 @@ func (r *DockerRuntime) setupEFSMounts(parentCtx context.Context, c runtimeTypes
 	return nil
 }
 
-// Setup listener
-func (r *DockerRuntime) setupTiniListener(ctx context.Context, c runtimeTypes.Container) (*net.UnixListener, error) {
-	if runtime.GOOS == "darwin" { //nolint:goconst
-		// On darwin (docker-for-mac), it is not possible to share
-		// darwin unix sockets with a linux guest container: https://github.com/docker/for-mac/issues/483
-		// Instead we gracefully degrade with a nil listener and move on
-		return nil, nil
-	}
-
-	fullSocketFileName := r.hostOSPathToTiniSocket(c)
-
+func (r *DockerRuntime) setupTiniListener(cName string) (*net.UnixListener, error) {
+	fullSocketFileName := r.hostOSPathToTiniSocket(cName)
 	l, err := net.Listen("unix", fullSocketFileName)
 	if err != nil {
 		return nil, err
 	}
-	unixListener := l.(*net.UnixListener)
 
+	unixListener := l.(*net.UnixListener)
 	err = os.Chmod(fullSocketFileName, 0777) // nolint: gosec
 	if err != nil {
 		return nil, err
 	}
 
-	go func() {
-		<-ctx.Done()
-		shouldClose(l)
-		if ctx.Err() == context.DeadlineExceeded {
-			log.WithField("ctxError", ctx.Err()).Error("Tini listener timeout occurred")
-		}
-	}()
+	return unixListener, nil
+}
+
+// setupTiniListeners sets up listening sockets in preperation for all the tinis from multiple
+// containers to connect to them
+func (r *DockerRuntime) setupTiniListeners(ctx context.Context, pod *v1.Pod) (map[string]*net.UnixListener, error) {
+	if runtime.GOOS == "darwin" { //nolint:goconst
+		// On darwin (docker-for-mac), it is not possible to share
+		// darwin unix sockets with a linux guest container: https://github.com/docker/for-mac/issues/483
+		// Instead we gracefully degrade with a nil listener and move on
+		return nil, nil
+	}
+
+	listeners := make(map[string]*net.UnixListener)
+
+	for _, c := range pod.Spec.Containers {
+		listener, err := r.setupTiniListener(c.Name)
+		if err != nil {
+			return nil, fmt.Errorf("Error while setting up tini listener for %s: %w", c.Name, err)
+		}
+		listeners[c.Name] = listener
+		go func() {
```

Comment:
Instead of having one goroutine per listener, why not have a single goroutine that loops over `listeners` once the context is cancelled?

Reply:
No particular reason?